codec-http2/src/main/java/io/netty/handler/codec/http2/InboundHttp2ToHttpAdapter.java

```diff
@@ -296,6 +297,9 @@ public void onPushPromiseRead(ChannelHandlerContext ctx, int streamId, int promi
             Http2Headers headers, int padding) throws Http2Exception {
         // A push promise should not be allowed to add headers to an existing stream
         Http2Stream promisedStream = connection.stream(promisedStreamId);
+        if (headers.status() == null) {
+            headers.status(CREATED.codeAsText());
```

Comment:
so created is more correct then ok ?

Reply:
@normanmaurer 
https://tools.ietf.org/html/rfc7540#section-8.2.1, 
The PUSH_PROMISE frame contains a header block that contains a complete set of request header fields that the server attributes to the request.
In fact, the first time I used OK as the Http response status, I changed the status to Created by referring to the test case (In the previous test case, we used 201 CREATED).

Reply:
Since it is not appropriate to define the response status on the rfc7540 spec, I think it is not important to be 200 or 201.

Reply:
According to the rfc7540 specification,

> https://tools.ietf.org/html/rfc7540#section-8.2.1
> Server push is semantically equivalent to a server responding to a
> request; however, in this case, that request is also sent by the
> server, as a PUSH_PROMISE frame.

> https://tools.ietf.org/html/rfc7540#section-8.2.2
> After sending the PUSH_PROMISE frame, the server can begin delivering
>  the pushed response as a response (Section 8.1.2.4) on a server-
>   initiated stream that uses the promised stream identifier.

Reply:
@normanmaurer The server sends the request headers in PUSH_PROMISE.
However, if you specify the response status in "onPushPromiseRead" to prevent NPE, 200 seems to be better than 201.
WDYT?